tests/data/PETPT.json

```diff
@@ -207,17 +207,17 @@
           "type": "variable"
         },
         {
-          "name": "eo_3",
+          "name": "eo_4",
           "type": "variable"
         },
         {
-          "name": "eo_2",
+          "name": "eo_3",
           "type": "variable"
         }
       ]
     },
     {
-      "name": "petpt__assign__eo_5",
+      "name": "petpt__assign__eo_6",
```

Comment:
It seems like we have introduced an indexing bug that is indexing all variables too quickly. We should only have `eo_0` through `eo_5` represented here, but instead we are seeing `eo_6`.

Reply:
@pratikbhd Looks too hard for me to fix - I'll leave it for you and close this PR :)